### PR TITLE
feat: add support for TypeScript 3.3 (and drop older versions)

### PR DESCRIPTION
**Pull request by IgorMinar**
https://blogs.msdn.microsoft.com/typescript/2019/01/31/announcing-typescript-3-3/

BREAKING CHANGE: TypeScript 3.1 and 3.2 are no longer supported.

Please update your TypeScript version to 3.3

### CODE REVIEW

**Comment by IgorMinar** [comment]
blocked on https://github.com/Microsoft/TypeScript/issues/29684

File: packages/compiler-cli/src/ngtsc/shims/src/host.ts [hunk 1]
```diff
@@ -84,7 +84,9 @@ export class GeneratedShimsHostWrapper implements ts.CompilerHost {
 
   getCurrentDirectory(): string { return this.delegate.getCurrentDirectory(); }
 
-  getDirectories(path: string): string[] { return this.delegate.getDirectories(path); }
+  getDirectories(path: string): string[] {
```
Comment (JoostK) [inline packages/compiler-cli/src/ngtsc/shims/src/host.ts]: I think the changes in #28055 (75f1a01846a91279002101faf63f3e11b87b7ed7) by @petebacondarwin are more accurate, as it won't cause host consumers (i.e. TypeScript itself) to think that the host does in fact has this method available. That would leave internal default behavior intact.

**Comment by JoostK** [comment]
Microsoft/TypeScript#29684 has been fixed in TS 3.3.3333.

**Comment by IgorMinar** [comment]
rebased... let's see how the CI likes all those threes :-)

**Comment by filipesilva** [comment]
@IgorMinar I fixed the remaining tests in https://github.com/angular/angular/pull/29004. You can take my extra fixup from there to this PR, or merge that PR.

**Comment by IgorMinar** [comment]
closing in favor of #29004. thanks @filipesilva

**Pull request closed without merging**